File: Search/maven/src/test/java/Search/RegexOperatorTest.java
package Search;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.Arrays;

public class RegexOperatorTest {
    
    RegexOperator regexOperator = new RegexOperator();

    
    @Test
    public void assortTheWordsTest(){

        String input = "It +is a +great -day +to -live ";
        String regex = "(\\+)(\\w*)" ; 

        ArrayList<String> actual = RegexOperator.assortTheWords(input, regex, 2);
        ArrayList<String> expected = new ArrayList <>(Arrays.asList("is", "great", "to"));

        assertEquals(expected, actual);
        
       
    }
    
}

File: Phase05/Search/Search.Test/ModelTest/RegexOperatorTest.cs
using Xunit;
using Search.Model;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Linq;

namespace Search.Test.ModelTest
{
    public class RegexOperatorTest
    {
        [Fact]
        public void AssortTheWordsTest()
        {
            var input = "It +is a +great -day +to -live ";
            var regex = new Regex("^(\\w*)");

            var actual = RegexOperator.AssortTheWords(input, regex, 1);
            var expected =  new string[]{ "It" }.ToList();

            Assert.Equal(actual,expected);

        }
    }
}